src/jsrm/utils/lie_operators.py

```diff
@@ -0,0 +1,271 @@
+            and the last three elements correspond to the linear component.
+
+    Returns:
+        Array: shape (6, 6)
+            A 6x6 matrix representing the adjoint transformation of the input screw vector.
+    """
+    vec6 = vec6.reshape(-1) # Ensure vec6 is a 1D array
+    if vec6.shape != (6,):
+        raise ValueError("Input vector must be of shape (6,)")
+    
+    ang = vec6[:3].reshape((3, 1))  # Angular as a (3,1) vector
+    lin = vec6[3:].reshape((3, 1))  # Linear as a (3,1) vector
+    
+    angtilde = tilde_SE3(ang)  # Tilde operator for angular part
+    lintilde = tilde_SE3(lin)  # Tilde operator for linear part
+    
+    adj = jnp.zeros((6, 6))
+    adj = adj.at[:3, :3].set(angtilde)  # Fill the top-left 3x3 block with the tilde operator of the angular part
+    adj = adj.at[3:, 3:].set(angtilde)  # Fill the bottom-right 3x3 block with the tilde operator of the angular part
+    adj = adj.at[3:, :3].set(lintilde)  # Fill the bottom-left 3x3 block with the tilde operator of the linear part
+    return adj
+
+@jit
+def hat_SE3(
+    vec6:Array
+)-> Array:
+    """
+    Computes the hat operator for a 6D vector of se(3).
+
+    Args:
+        vec (Array): array-like, shape (6,1)
+            A 6-dimensional vector representing the screw. 
+            The first three elements correspond to the angular component, 
+            and the last three elements correspond to the linear component.
+
+    Returns:
+        Array: shape (4, 4)
+            A 4x4 matrix representing the hat operator of the input screw vector.
+    """
+    vec6 = vec6.reshape(-1)  # Ensure vec6 is a 1D array
+    if vec6.shape != (6,):
+        raise ValueError("Input vector must be of shape (6,)")
+    
+    ang = vec6[:3].reshape((3, 1))  # Angular as a (3,1) vector
+    lin = vec6[3:].reshape((3, 1))  # Linear as a (3,1) vector
+    
+    angtilde = tilde_SE3(ang)  # Tilde operator for angular part
+    
+    hat = jnp.zeros((4, 4))
+    hat = hat.at[:3, :3].set(angtilde)  # Fill the top-left 3x3 block with the tilde operator of the angular part
+    hat = hat.at[:3, 3].set(lin.reshape((3,)))  # Fill the top-right column with the linear part
+    return hat
+
+@jit
+def Adjoint_g_SE3(
+    mat4:Array
+)-> Array:
+    """
+    Computes the adjoint representation of a 4x4 matrix.
+    
+    Args:
+        mat4 (Array): array-like, shape (4,4)
+            A 4x4 matrix representing the transformation.
+    
+    Returns:
+        Array: shape (4, 4)
+            A 4x4 matrix representing the Adjoint transformation of the input matrix.
+    """
+    ang = mat4[:3, :3]  # Extract the angular part (top-left 3x3 block)
+    lin = mat4[:3, 3].reshape((3, 1))  # Extract the linear part (top-right column)
+    
+    ltilde = tilde_SE3(lin)  # Tilde operator for linear part
+    
+    Adjoint = jnp.zeros((6, 6))
+    Adjoint = Adjoint.at[:3, :3].set(ang)  # Fill the top-left 3x3 block with the tilde operator of the angular part
```

Comment:
Please use `jnp.block` (https://docs.jax.dev/en/latest/_autosummary/jax.numpy.block.html)  instead of `.at[].set()`